awscli/customizations/s3/utils.py

```diff
@@ -278,22 +281,72 @@ def create_warning(path, error_message, skip_file=True):
 
 def find_chunksize(size, current_chunksize):
     """
-    The purpose of this function is determine a chunksize so that
-    the number of parts in a multipart upload is not greater than
-    the ``MAX_PARTS``.  If the ``chunksize`` is greater than
-    ``MAX_SINGLE_UPLOAD_SIZE`` it returns ``MAX_SINGLE_UPLOAD_SIZE``.
+    The purpose of this function is determine a chunksize so that the number of
+    parts in a multipart upload is not greater than the ``MAX_PARTS``, and that
+    the chunksize is not less than the minimum chunksize.
+
+    :param size: The size of the file to upload
+    :param current_chunksize: The currently configured chunksize
+    :return: If the given chunksize is valid, it is returned. Otherwise a valid
+        chunksize is calculated and returned.
+    """
+    size = adjust_chunksize_for_max_parts(size, current_chunksize)
+    return adjust_chunksize_to_upload_limits(size)
```

Comment:
For a pure API standpoint, I would expect a utility function like this to raise an exception if the constraints can't be met (instead of giving back a value that something somewhere else should reject), but I would be ok with documentation/comments explaining this for now.